[FIX][website_event] Make registration form usable in small screens.

Yajo opened the pull request:
Description of the issue/feature this PR addresses: Fix the small screen layout for event registrations.

Current behavior before PR: Before this patch, the event registration form was almost impossible to use in a phone.

Desired behavior after PR is merged: Sane form layout both in small and big screens.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
@Tecnativa

pedrobaeza commented:
Can you put some screenshots of before/after for clarifying?

Yajo commented:
This needs a little bit further patching, tomorrow I'll update it. WIP for now.

Yajo commented:
This is WIP no more.

Let me show the results. This is **before** the patch:

![captura de pantalla de 2017-02-14 11-44-12](https://cloud.githubusercontent.com/assets/973709/22926127/a52b4a04-f2ab-11e6-973d-094e920788b3.png)

![captura el 2017-02-14 a las 11 44 38](https://cloud.githubusercontent.com/assets/973709/22926145/ae5446da-f2ab-11e6-80cb-d77a983bba94.png)

![captura el 2017-02-14 a las 11 44 46](https://cloud.githubusercontent.com/assets/973709/22926152/b7d2de6a-f2ab-11e6-9457-a122f79562eb.png)

As you can see, the mobile one is almost completelly useless.

Now, **after** the patch:

![captura de pantalla de 2017-02-14 11-45-28](https://cloud.githubusercontent.com/assets/973709/22926180/d194da88-f2ab-11e6-800c-966caeb27892.png)

![captura el 2017-02-14 a las 11 45 45](https://cloud.githubusercontent.com/assets/973709/22926187/d7f802ce-f2ab-11e6-94cd-f5bd2ac05781.png)

![captura el 2017-02-14 a las 11 48 34](https://cloud.githubusercontent.com/assets/973709/22926189/d973c408-f2ab-11e6-8be0-af1c2096c679.png)

Now the buy section has a bigger visual impact, XML templates are structured in a sane and reusable way, and --most important-- you can actually use it with a phone.

pedrobaeza commented:
This looks better of course. Maybe the blue color is argueable. Why the runbot error?

Yajo commented:
Seems like I broke a Tour. I fixed it now.

I added the blue colour because this page is full of similar panels, and this way we give some higher visual impact to the registration form, which is the one that converts more directly in money. However it's not the main issue here, so it can be changed if required.

Not sure who to ping here... @Yenthe666?

pedrobaeza commented:
Uhm, me neither.

Yenthe666 commented:
Quite sure it is @jke-be you need. He is usually here for website and JS stuff.

pedrobaeza commented:
This is conflicting with current branch. Can you check?

Yajo commented:
Conflicts were introduced by @mpa-odoo in https://github.com/odoo/odoo/commit/5d4dd98af2672ccdde37d9587653c80a0ec7c661. I solved them now.

His improvements have been kept unless one little detail: [He displayed "Register now" if only free tickets were available and "Order now" if there was any paid ticket in the list](https://github.com/odoo/odoo/commit/5d4dd98af2672ccdde37d9587653c80a0ec7c661#diff-d4523a5be609dc145d4a2533bf9008eaR96). It seems to me that "Register now" is useful for both scenarios, so I'm using only that.

The rest of the logic should stay the same while having an easier to maintain/reuse template and, most important, usable in phones.

pedrobaeza commented:
@mpa-odoo, maybe you can check also this improvement?

@JKE-be, can you also check this fine one?

JKE-be commented on addons/website_event_sale/views/event_templates.xml:
```diff
@@ -19,90 +19,66 @@
-                              "from_currency": event.company_id.sudo().currency_id,
-                              "display_currency": website.get_current_pricelist().currency_id
+    <xpath expr="//t[@t-id='tickets']" position="replace">
+        <t t-set="saleable_tickets" t-value="event.event_ticket_ids.filtered(lambda t: not t.is_expired"/>
```
missing one parenthesis

JKE-be commented on addons/website_event_sale/views/event_templates.xml:
```diff
@@ -19,90 +19,66 @@
-        </button>
-    </xpath>
-    <xpath expr="//form[@id='registration_form']" position="attributes">
-        <attribute name="t-if">event.event_ticket_ids</attribute>
```
why remove it ?

JKE-be commented:
Landed in master odoo/odoo@e7fdd6d

pedrobaeza commented:
Thank you very much!! Very appreciated.

I suppose that you have fixed yourself your comments.
